API_doc/Libconfig.h

```diff
@@ -0,0 +1,53 @@
+//
+// Must be included to use the LibConfig
+//
+//
+
+#pragma once
+
+# include <stddef.h>
+
+#ifdef _cplusplus
+namespace LibConfig
+{
+#endif
+
+  typedef void (*onChangeNotifier)(void *data, char const *newValue);
+
+  /*
+   * contain a unique id for each configuration
+   */
+  struct Id {
+    void *data;
+    size_t size;
```

Comment:
Needs comments explaining the fields exactly (such as the fact that `size` is the size of `data` in bytes).